storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/RocksDbDatabase.java

```diff
@@ -252,6 +254,32 @@ public void update(final StorageUpdate event) {
     return finalizedDao.streamFinalizedBlocks(startSlot, endSlot);
   }
 
+  @Override
+  public List<Bytes32> getStateRootsBeforeSlot(final UnsignedLong slot) {
+    return hotDao.getStateRootsBeforeSlot(slot);
+  }
+
+  @Override
+  public void addHotStateRoot(final Bytes32 stateRoot, final SlotAndBlockRoot slotAndBlockRoot) {
+    try (final HotUpdater updater = hotDao.hotUpdater()) {
+      updater.addHotStateRoot(stateRoot, slotAndBlockRoot);
+      updater.commit();
+    }
+  }
```

Comment:
This doesn't feel right - I'd have expected new hot state roots to be added as part of a`StorageUpdate` event since they'll only be added as part of importing a block and thus should be stored in the same transaction as the rest of the updates.

In any case, we definitely wouldn't want to store them one at a time - it's much more efficient to store them in a batch.

Ultimately though this doesn't affect the disk layout so we could start with this to make these tests work then move to something else in later PRs when we're actually capturing the data during state transition.

Reply:
yep agreed, this shows store, purge, retrieve, and we can work out the messy details once we actually start storing data I guess.